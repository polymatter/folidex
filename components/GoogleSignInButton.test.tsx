import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import GoogleSignIn from './GoogleSignInButton';

interface Snapshot {
  children: Array<{ props: { source?: string } }>
}

describe("<GoogleSignIn>", () => {

  test("", () => {
    const { toJSON } = render(<GoogleSignIn />);

    const snapshot = toJSON() as Snapshot;
    // displays sign in with google image
    // TODO: Can not test for a particular image as require returns a number (see the snapshot). This test "works" but I'm not sure its meaningful so its commented out
    // expect(snapshot.children.find(c => c.props.source === require('../assets/images/SignInWithGoogle.png'))).not.toBeUndefined();

    expect(snapshot).toMatchSnapshot(); 
  })

})