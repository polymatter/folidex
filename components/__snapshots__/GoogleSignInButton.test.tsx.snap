// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleSignIn>  1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Image
    source={1}
    style={
      Object {
        "height": 38,
        "width": 180,
      }
    }
  />
</View>
`;
